### ISSUE TITLE
Version 8.1.3 break when a large table has a colspan in it

### ISSUE THREAD

**Issue by nd4c** [mpdf/mpdf#1777]
### Guidelines

- [X] I understand that [if I fail to provide all required details, this issue may be closed without review](https://github.com/mpdf/mpdf/blob/development/.github/CONTRIBUTING.md).

### Description of the bug

When a table spans more than one page and at least one `<td>` or `<th>` has a colspan attribute, a "New Page" is placed after most `</tr>` tags until the last few fit on a page.  There are some discrepencies - some pages will display two rows, not just one.

### mPDF version

8.1.3

### PHP Version and environment (server type, cli provider etc., enclosing libraries and their respective versions)

PHP 8.0.26 on Apache 2.4.54 on LInux Mint

### Reproducible PHP+CSS+HTML snippet suffering by the error

```
<?php

require_once __DIR__ . '/vendor/autoload.php';

$config = [
	'mode' => 'UTF-8',
	'format' => 'Letter',
	'default_font' => 'Arial',
	'tempDir' => '../runtime/tmp',
	'shrink_tables_to_fit' => 1,
];

$mpdf = new \Mpdf\Mpdf($config);

// The colspan="2" seems to cause the problem
$html = [
	'<h4>Hello world!</h4>',
	'<div>',
	'<table>',
	'<thead>',
	'<tr>',
	'<th>header</th>',
	'<th>header</th>',
	'<th>header</th>',
	'<th>header</th>',
	'<th>header</th>',
	'<th>header</th>',
	'<th colspan="2">header</th>',
	'</tr>',
	'</thead>',
	'<tbody>',
];

// The last 40 will print on a single page, but first page has two line for the Hello world, so only 38 for first page.
$rows = 80;
while ($rows--) {
	$html[] = "<tr>";
	$html[] = "<td>Some Wide Date $rows</td>";
	$html[] = "<td>Some Wide Date $rows</td>";
	$html[] = "<td>Some Wide Date $rows</td>";
	$html[] = "<td>Some Wide Date $rows</td>";
	$html[] = "<td>Some Wide Date $rows</td>";
	$html[] = "<td>Some Wide Date $rows</td>";
	$html[] = "<td colspan=\"2\">Some Wide Date $rows</td>";
	$html[] = "</tr>";
}

$html[] = '</tbody>';
$html[] = '</table>';
$html[] = '</div>';

$mpdf->WriteHTML(implode(PHP_EOL, $html));
$mpdf->Output('test.pdf', 'I');

exit();

```

**Comment by nd4c** [comment]
#1776 was closed.  Did not resolve problem.  By the way, version 8.1.2 has no problems with the colspan in long tables.

**Comment by nd4c** [comment]
@finwe , Sorry I opened a new issue, I couldn't find a way to re-open the 1776 one.  However, the description of the problem is more correct in this issue.  Thanks for looking into this.

**Comment by finwe** [comment]
Yeah, it's okay. I remember there was a pull request regarding colspans, I'll probably revert it and reopen if I cannot find a solution quickly.

**Comment by nd4c** [comment]
After some further investigation, it looks like a colspan in a `<th>` does not cause any issues, only a colspan in a `<td>`.

Also, note that the issue is not consistent.  On either page 8 or 9 it will display two rows, not one, but I haven't found anything in particular that "controls" that.

**Comment by nd4c** [comment]
I found that the number of columns affects which page will display two rows.  

Assuming one column has a colspan of 2 and ignoring the last page:
1 column: it never shows two rows 
2 columns: page 3
3 columns: page 4
etc.

**Comment by nd4c** [comment]
I believe the error is cause by line 20813 in src/Mpdf.php

was (at line 20796 in 8.1.2):
				if ($table['cells'][$i][$j]) {

is now (at line 20813 in 8.1.3):
				if (isset($table['cells'][$i][$j])) {

Got it to work by changing that line to:
				if (!empty($table['cells'][$i][$j])) {

**Comment by nd4c** [comment]
I found the error:

Line 20813 in src/Mpdf.php it says:
				`if (isset($table['cells'][$i][$j])) {`

Change that to:
				`if (!empty($table['cells'][$i][$j])) {`

That worked for me.  Not sure if it breaks anything else...

**Comment by finwe** [comment]
@nd4c Thanks for your review and a proposed fix! Released as v8.1.4

**Comment by HikeGrig** [comment]
> I believe the error is cause by line 20813 in src/Mpdf.php
> 
> was (at line 20796 in 8.1.2): `if ($table['cells'][$i][$j]) {`
> 
> is now (at line 20813 in 8.1.3): `if (isset($table['cells'][$i][$j])) {`
> 
> Got it to work by changing that line to: `if (!empty($table['cells'][$i][$j])) {`
> 
> I have no idea if this breaks anything else...

Thanks a lot for your answer, this saved me a lot of time.
